playwright/_impl/_connection.py

```diff
@@ -357,6 +359,8 @@ def dispatch(self, msg: ParsedMessagePayload) -> None:
             callback = self._callbacks.pop(id)
             if callback.future.cancelled():
                 return
+            if callback.no_reply:
```

Comment:
What was bad about calling set_result on the callback future?

Reply:
See the PR description, that we received an error (waitForEventInfo after), were setting it on the protocol callback but never awaiting it, hence it was showing the warning in the console.